main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"log"
+	"regexp"
+	"strconv"
+	"sync"
+)
+
+func Serve(w http.ResponseWriter, r *http.Request) {
```

Comment:
Maybe I'm missing context here, but unless you have a good reason to regex match URL paths, I think you should probably use something like net/http.ServeMux to register different, multiple handlers instead of collapsing into a single handler you match the path against.

Your number of cases is small here, but imagine you had 80 paths you needed to handle. Using ServeMux would immediately match against the pattern (very roughly like a hashmap in O(1) time). Your current implementation has to match against each and is average and worst case O(n) time w.r.t. a regex operation.